lib/matplotlib/colors.py

```diff
@@ -943,6 +943,23 @@ def _set_extremes(self):
             self._lut[self._i_over] = self._lut[self.N - 1]
         self._lut[self._i_bad] = self._rgba_bad
 
+    def with_alpha(self, alpha):
+        """
+        Return a copy of the colormap with a new uniform transparency.
+
+        Parameters
+        ----------
+        alpha : float
+             The alpha blending value, between 0 (transparent) and 1 (opaque).
+        """
+        if not 0 <= alpha <= 1:
```

Comment:
I think this is going to return `ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()` is someone inputs an array.  
```suggestion
        if (len(alpha) > 1) or (not 0 <= alpha <= 1):
```

Reply:
Your observation on the error is correct. What's wrong with that? / What would you expect? I can't make sense of the suggestion, because it would raise on floats ("object of type 'float' has no len()").

Reply:
It is cryptic if someone passes an utterable.  You probably want something like:

```
if (isinstance(x, Iterable) and hasattr(x, "__len__") and len(alpha) > 1):
  raise(ValueError, 'alpha must be a single value, not a list or array')
```

Reply:
Generally, we don't jump extra hoops to catch types that are not supported and give them a nicer error message.

But granted, `Artist.set_alpha` does something similar, so I've taken the same approach as there.